src/models/classes/container.js

```diff
@@ -38,6 +38,44 @@ export default class Container extends StorageCore
      */
     removeItem(product)
     {
+        // Maken an array of the perish time and corresponding product name, which can be compared later.
+        let perishTimeArray = [];
+        let perishNameArray = [];
+        let perishDuplicates = {};
+        for (let i = 0; i < this.items.length; i++) {
+
+           if (this.items[i].values.isPerishable) {
+
+               perishTimeArray.push(this.items[i].values.perishable);
+               perishNameArray.push(this.items[i].name);
+           }
+        }
+
+        // https://stackoverflow.com/a/18417854
+        // Creates an object with the duplicate name,
+        // and their index in the original array.
+
+        Array.prototype.getDuplicates = function () {
```

Comment:
Not really problematic, but don't do this here. You're extending the Array prototype in a place where this is _not_ expected. Better to do this in main.js, by including a new file there called e.g. `extensions.js`